wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/Util.kt

```diff
@@ -90,3 +96,55 @@ expect interface MutableQueue<T : Any> : MutableCollection<T> {
 }
 
 internal expect fun <T : Any> mutableQueueOf(): MutableQueue<T>
+
+// TODO internal and friend for wire-compiler: https://youtrack.jetbrains.com/issue/KT-20760
+/**
+ * Replace types in this schema which are present in [typesToStub] with empty shells that have no
+ * outward references. This has to be done in this module so that we can access the internal
+ * constructor to avoid re-linking.
+ */
+fun Schema.withStubs(typesToStub: Set<ProtoType>): Schema {
+  if (typesToStub.isEmpty()) {
+    return this
+  }
+  return Schema(protoFiles.map { protoFile ->
```

Comment:
This is the only change in this PR aside from code relocation. We call `Schema`'s internal constructor rather than `Schema.fromFiles` (which links).